<?xml version="1.0"?>
<!--Only use the left camera-->
<launch>
<!--  <arg name="image_in" default="/vizzy/l_camera/image_rect_color" /> -->

  <arg name="image_in" default="/vizzy/r_camera/image_raw" />
  <arg name="image_out" default="pedestrian_detector_image" />
  <arg name="detector_type" default="full" />
  <arg name="camera" default="r_camera" />
  <arg name="follower" default="true" />
  <arg name="following_planner_mode" default="global_map" />
  
  <node pkg="pedestrian_detector" type="detector" name="detector" output="screen">
    <remap from="image_in" to="$(arg image_in)"/>
    <remap from="image_out" to="$(arg image_out)"/>
    <param name="detector_type" type="str" value="$(arg detector_type)"/>
    <param name="camera" type="str" value="$(arg camera)"/>
    <rosparam file="$(find pedestrian_detector)/config/detector_params.yaml" command="load"/>
  </node>
  <node pkg="pedestrian_detector" type="tracker" name="tracker" output="screen">
	<rosparam file="$(find pedestrian_detector)/config/tracker_params.yaml" command="load"/>
  </node>
  <group if="$(arg follower)">
  <node pkg="pedestrian_detector" type="follower" name="follower" output="log">
	<remap from="twist_topic" to="/vizzy/twist2nav2d"/>
	  <rosparam file="$(find pedestrian_detector)/config/follower_params_$(arg following_planner_mode).yaml" command="load"/>
  </node>
  </group>
</launch>
